### PR TITLE
xsdk-examples: use 'parallel=False' with tests - as they may fail with a large '-j' [when used with +cuda]

### PR DESCRIPTION
**Pull request by balay**

### CODE REVIEW

**Comment by v-dobrev** [comment]
It is strange that Spack will run tests in parallel -- I'd expect many HPC packages do run parallel tests, so it would make sense to execute the tests one by one. With that in mind, using `-j 1` maybe even better than `-j 2`.

**Review by v-dobrev — DISMISSED** [review]
Maybe use `-j 1` instead of `-j 2`?

**Comment by adamjstewart** [comment]
`parallel=False` is better than `'-j1'` because the latter will actually run `make test -j16 -j1` while the former will run `make test`.

**Comment by balay** [comment]
Ok - updated PR to use `parallel=False`

**Review by v-dobrev — DISMISSED** [review]

File: var/spack/repos/builtin/packages/xsdk-examples/package.py [hunk 1]
```diff
@@ -92,3 +92,8 @@ def cmake_args(self):
             args.append('-DZLIB_LIBRARY_DIR=%s' % spec['zlib'].prefix.lib)
 
         return args
+
+    def check(self):
+        if self.run_tests:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xsdk-examples/package.py]: You can remove this line, `check` is only run when this is true

Comment (balay) [reply var/spack/repos/builtin/packages/xsdk-examples/package.py]: Ok done.

The prior format was a copy/paste from current usage in other packages.

**Review by v-dobrev — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**